### ISSUE TITLE
re-clustering a cell subset but cell identity numbers are not completely showing up 

### ISSUE THREAD

**Issue by Leran10** [satijalab/seurat#4967]
Hi,

After running T-SNE plot on dataset, I tried to extract a specific cell type then do reclustering. But I found that the some of the cell identity IDs are skipped in the reclustered T-SNE plot (0,2,3,4,6... instead of 0,1,2,3,4,5,6):

![image](https://user-images.githubusercontent.com/17597071/130263972-8d2defe3-bfa6-4888-ae87-62d229f3e5a8.png)

Here is my code:

```
DefaultAssay(combined.all) <- "integrated"

# subset Seurat to only Epithelial cells
subset.Epithelial <- subset(x = combined.all, idents = "Epithelial cells")
# run tsNE
subset.Epithelial <- RunTSNE(object = subset.Epithelial)
# run umap
subset.Epithelial <- RunUMAP(subset.Epithelial, dims = 1:20) # reduction = "pca"

# T-sne plot
Epithelial.clusters.tsne <- DimPlot(subset.Epithelial, reduction = "tsne",group.by = "seurat_clusters",label = TRUE)
```

I cannot figure this out nor find help from other website resources. Could you provide some help on how to interpret this?

Thanks so much!

**Comment by samuel-marsh** [comment]
Hi,

Can you please post the exact code that you ran for the subsetting and the reclustering before making this plot?

Thanks,
Sam

**Comment by Leran10** [comment]
Hi @samuel-marsh, Sorry I forgot that, thanks for the reminding! code posted!

Leran

**Comment by samuel-marsh** [comment]
Hi Leran,

So it's unclear to me why there are numbers at all if you had renamed all of the clusters using cell names prior to subsetting (i.e. epithelial cells).  However, after subsetting it is not sufficient to simply rerun the tsne and umap functions as the info that those functions draw from (i.e. SNN, pca, etc) are all still based on the full object before subsetting.

If you are interested in getting finer structure by subsetting I would suggest rerunning the entire process beginning with finding variable genes/scale data (or SCTransform depending on what pipeline you are using) followed by PCA, neighbors, clusters, and finally dimreduction before plotting again.  This should solve issue I would think.

Best,
Sam

**Comment by torkencz** [comment]
I agree with @samuel-marsh . You should rerun the all of the normalization steps before running tSNE on the subsetted object.

**Comment by Leran10** [comment]
Thanks @samuel-marsh,

So I have assigned cell names to the clusters that generated from the entire dataset, that's why you saw I had the epithelial cells. Then like what you said I want to further look into the epithelial cells to get better resolution, so I extract this cell and do the clustering again.

And thanks so much for the instruction! I can understand that I need to recalculate some scores after subsetting, but I'm not sure about about the normalization step NormalizeData() and scaling step ScaleData(). Since I have already scaled the data at the beginning, If I run these steps again, would that cause trouble? Unless it is running based on the raw counts but not the already normalized + scaled counts? 

Thank you!
Leran
